rcl/test/rcl/test_context.cpp

```diff
@@ -128,4 +128,26 @@ TEST_F(CLASSNAME(TestContextFixture, RMW_IMPLEMENTATION), nominal) {
 TEST_F(CLASSNAME(TestContextFixture, RMW_IMPLEMENTATION), bad_fini) {
   EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, rcl_context_fini(nullptr));
   rcl_reset_error();
+
+  rcl_init_options_t init_options = rcl_get_zero_initialized_init_options();
+  rcl_ret_t ret = rcl_init_options_init(&init_options, rcl_get_default_allocator());
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    EXPECT_EQ(RCL_RET_OK, rcl_init_options_fini(&init_options)) << rcl_get_error_string().str;
+  });
+
+  rcl_context_t context = rcl_get_zero_initialized_context();
+  ret = rcl_init(0, nullptr, &init_options, &context);
+  EXPECT_EQ(RCL_RET_OK, ret);
+
+  ret = rcl_context_fini(&context);
+  EXPECT_EQ(ret, RCL_RET_INVALID_ARGUMENT);
+  rcl_reset_error();
+
+  ret = rcl_shutdown(&context);
+  EXPECT_EQ(ret, RCL_RET_OK);
+
+  ret = rcl_context_fini(&context);
+  EXPECT_EQ(ret, RCL_RET_OK);
```

Comment:
are changes in this file related?

Reply:
Yes, they are. This patch changes `rcl_timer_init` API to accept an `rcl_timer_options_t *` argument, so all call sites have to be updated.

---

Alternatively, we could either have a separate `rcl_timer_init_with_options()` overload (though that doesn't scale all that well) OR use macros for versioning and deprecation. In general, I don't think we have a good policy when breaking C API.